lib/client.js

```diff
@@ -60,6 +60,16 @@ var clientOptions = require('./client-options');
  *  Setting it to false will cause less overhead and the reference of the network buffer to be maintained until the row / result set are de-referenced.
  *  Default: true.
  * </p>
+ * @property {Boolean} encoding.useUndefinedAsUnset Valid for Cassandra 2.2 and above. Determines that, if a parameter is set to
```

Comment:
Would it be worth validating on initialization of Client that if protocolVersion < 4 is used and useUndefinedAsUnset is specified as true that we fail fast?  We would probably want to do it when we detect the max allowed protocol version as well.  I think this could be useful, but I can see it not being necessary.